dojo/group/queries.py

```diff
@@ -44,8 +45,26 @@ def get_authorized_group_users(permission):
     if user.is_staff and settings.AUTHORIZATION_STAFF_OVERRIDE:
         return Dojo_Group_User.objects.all()
 
-    if hasattr(user, 'global_role') and role_has_permission(user.global_role.role.id, permission):
+    if hasattr(user, 'global_role') and user.global_role.role is not None and role_has_permission(user.global_role.role.id, permission):
         return Dojo_Group_User.objects.all()
 
     groups = get_authorized_groups(permission)
-    return Dojo_Group_User.objects.filter(dojo_group__in=groups)
+    return Dojo_Group_User.objects.filter(group__in=groups)
+
+
+def get_authorized_group_users_for_user(user):
+    groups = get_authorized_groups(Permissions.Group_View)
+    groups = Dojo_Group_User.objects.filter(user=user, group__in=groups)
```

Comment:
This would be even better if we sort by group name.

```suggestion
    groups = Dojo_Group_User.objects.filter(user=user, group__in=groups).order_by('group__name')
```